src/test/java/test/thread/Github1636Sample.java

```diff
@@ -0,0 +1,29 @@
+package test.thread;
+
+import org.testng.annotations.Test;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class Github1636Sample {
+    private static Map<Long, Boolean> map = new ConcurrentHashMap<>();
+    static Set<Long> threads = Collections.newSetFromMap(map);
```

Comment:
For my information, why do you create a set from a map? Concurrent reason?

Reply:
@juherr - Yes. Java doesn't have a `ConcurrentHashSet`. The recommended way of doing it is by retrieving the keys from a  `Map`.